lib/cache/cache_test.go

```diff
@@ -2038,6 +2062,72 @@ func TestDatabases(t *testing.T) {
 	require.Equal(t, 0, len(out))
 }
 
+func TestRelativeExpiry(t *testing.T) {
```

Comment:
This test takes over 20 seconds. Can we move it to integration instead? I really want to get to the point where `make test-go` completes in a reasonable amount of time, and adding more long-running tests moves us in the wrong direction.

Reply:
This test relies on a lot of package-local stuff.  Moving it would be non-trivial and is definitely not something I'd want to do in a backport.

IMO moving tests to `integration` just because they take a while is probably not ideal.  Its fairly common for unit tests to involve package-local fields/helpers/etc.  This is a good thing, since it helps us limit our public API surface (and helps reduce final binary size by letting the compile factor out things it knows are unused).

If we want a fast test command, maybe we could look into adding some flags that skip longer running unit tests?

Reply:
Can we track a follow-up item to speed this up, break it apart, or otherwise parallelize it? 20 seconds is outrageous for a unit test, and incentivizes people to simply not run them.